app/models/commits_and_pushes.rb

```diff
@@ -1,49 +1,55 @@
+# frozen_string_literal: true
+
 class CommitsAndPushes < ActiveRecord::Base
   fields do
-    no_jira :boolean, default: false, null: false
+    no_jira :boolean, default: false
   end
 
   include ErrorsJson
 
-  ERROR_ORPHAN_NO_JIRA_ISSUE_NUMBER = 'orphan_no_jira_issue_number'.freeze
-  ERROR_ORPHAN_JIRA_ISSUE_NOT_FOUND = 'orphan_jira_issue_not_found'.freeze
-  NO_JIRA_FOUND                     = 'No-Jira tag found'.freeze
-
-  belongs_to :push, inverse_of: :commits_and_pushes, required: true
-  belongs_to :commit, inverse_of: :commits_and_pushes, required: true
+  ERROR_ORPHAN_NO_JIRA_ISSUE_NUMBER = 'orphan_no_jira_issue_number'
+  ERROR_ORPHAN_JIRA_ISSUE_NOT_FOUND = 'orphan_jira_issue_not_found'
+  NO_JIRA_FOUND                     = 'No-Jira tag found'
 
-  scope :for_push, lambda { |push| where(push: push) }
-  scope :with_no_jira_tag, -> { where(no_jira: true) }
+  belongs_to :push,   inverse_of: :commits_and_pushes, optional: false
+  belongs_to :commit, inverse_of: :commits_and_pushes, optional: false
 
-  def self.create_or_update!(commit, push, error_list = nil)
-    record = CommitsAndPushes.where(commit: commit, push: push).first_or_initialize
-    # preserve existing errors if not specified
-    if error_list
-      record.error_list = error_list
+  class << self
+    def with_no_jira_tag
+      where(no_jira: true)
     end
-    # if this is a newly created relationship, copy the ignore flag from the most recent relationship
-    unless record.id
-      record.copy_ignore_flag_from_most_recent_push
+
+    def create_or_update!(commit, push, error_list = nil)
+      CommitsAndPushes.find_or_initialize_by(commit: commit, push: push).tap do |record|
+        # preserve existing errors if not specified
+        if error_list
+          record.error_list = error_list
+        end
+        # if this is a newly created relationship, copy the ignore flag from the most recent relationship
+        unless record.id
+          record.copy_ignore_flag_from_most_recent_push
+        end
+        record.save!
+      end
     end
-    record.save!
-    record
-  end
 
-  def self.get_error_counts_for_push(push)
-    get_error_counts(with_unignored_errors.where(push: push))
-  end
+    # TODO: this can be cleaned up with scoping and Associa
```

Comment:
Truncated comment?

Reply:
Yea, good catch. I actually just went ahead and took care of this in 99a2d914ff1070c90d218299605594c3b692ce8f so that this comment isn't necessary.